### PR TITLE
read argparse default from git config

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: git_pull_request/__init__.py [hunk 1]
```diff
@@ -102,6 +103,26 @@ def git_remote_url(remote="origin"):
...
+    out = _run_shell_command(
+        ["git", "config", "--get", "git-pull-request." + option],
+        output=True, raise_on_error=False)
+    return out or default
+
+
+def git_config_add_argument(parser, option, *args, **kwargs):
+    default = kwargs.get("default")
+    isboolean = kwargs.get("action") in ["store_true", "store_false"]
+    if isboolean and not default:
```
Comment (jd) [inline git_pull_request/__init__.py]: maybe `if isboolean and default is None` would be more explicit

File: git_pull_request/__init__.py [hunk 2]
```diff
@@ -542,36 +563,40 @@ def main():
...
+                            action="store_true",
+                            help="Don't rebase branch before pushing.")
+    git_config_add_argument(
+        parser,
         "--force-editor",
         action="store_true",
         default=False,
         help="Force editor to run to edit pull-request message.")
-    parser.add_argument(
+    git_config_add_argument(
```
Comment (jd) [inline git_pull_request/__init__.py]: I don't see what comment would make sense by default in git config?

Comment (sileht) [reply git_pull_request/__init__.py]: Done

File: git_pull_request/__init__.py (hunk 1 above)
Comment (sileht) [reply git_pull_request/__init__.py]: Done

**Pull request merged**